assets/js/custom/Translation.js

```diff
@@ -1,7 +1,121 @@
 /* eslint-env jquery */
 
 // eslint-disable-next-line no-unused-vars
-function Translation (hasDescription, hasCredit) {
+function Translation (hasDescription, hasCredit, translatedByText) {
+  const providerMap = {
+    itranslate: 'iTranslate'
+  }
+  let languageMap = {}
+  let targetLanguage = document.documentElement.lang
+
+  $(document).ready(function () {
+    setTargetLanguage()
+    getLanguageMap()
+  })
+
+  $(document).on('click', '.comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('comment-translation-button-'.length)
+
+    $(this).hide()
+
+    if (isTranslationNotAvailable(commentId)) {
+      $('#comment-translation-loading-spinner-' + commentId).show()
+      translateComment(commentId)
+    } else {
+      openTranslatedComment(commentId)
+    }
+  })
+
+  $(document).on('click', '.remove-comment-translation-button', function () {
+    const commentId = $(this).attr('id').substring('remove-comment-translation-button-'.length)
+    $(this).hide()
+    $('#comment-translation-button-' + commentId).show()
+    $('#comment-translation-wrapper-' + commentId).slideUp()
+    $('#comment-text-wrapper-' + commentId).slideDown()
+  })
+
+  function setTargetLanguage () {
+    const decodedCookie = decodeURIComponent(document.cookie).split(';')
+    for (let i = 0; i < decodedCookie.length; i++) {
+      if (decodedCookie[i].includes('hl=') && decodedCookie[i].length < 10) {
+        targetLanguage = decodedCookie[i].substring(decodedCookie[i].indexOf('=') + 1).replace('_', '-')
+      }
+    }
+  }
+
+  function getLanguageMap () {
+    $.ajax({
+      url: '../languages',
+      type: 'get',
+      success: function (data) {
+        languageMap = data
+      }
+    })
+  }
+
+  function isTranslationNotAvailable (commentId) {
+    return $('#comment-text-translation-' + commentId).attr('lang') !== targetLanguage
+  }
+
+  function openGoogleTranslatePage (commentId) {
+    const text = document.getElementById('comment-text-' + commentId).innerText
+    window.open(
+      'https://translate.google.com/?q=' + encodeURIComponent(text) + '&sl=auto&tl=' + targetLanguage,
+      '_self'
+    )
+  }
+
+  function setSourceAndTargetLanguage (commentId, firstLang, secondLang, firstLangMapped, secondLangMapped) {
+    const transition = translatedByText.substring(translatedByText.indexOf(firstLang) + firstLang.length, translatedByText.indexOf(secondLang))
+
+    $('#comment-translation-credit-transition-' + commentId).text(transition)
+    $('#comment-translation-first-language-' + commentId).text(firstLangMapped)
+    $('#comment-translation-second-language-' + commentId).text(secondLangMapped)
+  }
+
+  function setTranslatedCommentData (commentId, data) {

```

Comment:
Will this work if provider is after source and target language?

Reply:
I was thinking about this edge case. I think I will refactor this section of the code when I implement the selectors to split it up so that it is "Before first language" - "first language" - "between languages" - "second language" - "after second language". Then, the provider will be replaced in any of the "Before first language", "between languages", or "after second language". So, I will refactor this whole section when the selectors are implemented (hopefully next week) and fix all the things you pointed out if that's okay.

Reply:
Updated the translated by line to handle all orders of source and target language